### PR TITLE
Navbars for everybody!, bulk actions, and quick notes

### PR DESCRIPTION
**Pull request by mattharris5**
The main feature additions in this PR are:

* Navbars for everybody, not just admins. Unprivileged users see three items:
  * "My mentees" - instant access to their assigned mentees (cached)
  * "Participants" - mostly for searching, and adding a new mentee
  * "High Schools" - easy access to the three action buttons for their assigned high schools (All Students, Take Attendance, Statistics)
* Bulk actions, which activate if you select one or more participants from a list. In the sidebar you'll see the bulk actions appear (with light blue background) and include the following tasks, for now:
  * Send email - which constructs a custom `mailto:` redirect
  * Assign to mentor
  * Flag for followup
  * Add note
* Quick notes/followup, which makes it easy to add a note or flag a participant for follow-up by clicking on the little orange flag in the table row. If they don't have a follow-up flag already assigned, a semi-transparent one will show up to show the user where to click.

### CODE REVIEW

File: app/models/people/person.rb [hunk 1]
```diff
@@ -383,7 +388,7 @@ def background_check_pending?
   # to the text "OK". 
   def passed_sex_offender_check?
     return false if sex_offender_check_result.nil?
-    valid_length = Customer.current_customer.background_check_validity_length
+    valid_length = Customer.current_customer.background_check_validity_length || 0
```
Comment (mitchellharper12) [inline app/models/people/person.rb]: Sorry about that :/

File: app/controllers/participant_bulk_actions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  before_filter :check_authorization
+  
+	def send_email
+		@emails = @participants.collect(&:email).flatten.compact.uniq
+		if @emails.empty?
+			flash[:error] = "You must select at least one record with an e-mail address."
+			render :text => flash[:error], :status => 200
+		else
+      # flash[:notice] = "Sent #{@template.pluralize(@emails.count, "e-mail address")} to your e-mail program."
+			render :js => "window.location.href = 'mailto:#{@emails.join(",")}';"
```
Comment (mitchellharper12) [inline app/controllers/participant_bulk_actions_controller.rb]: I'm not sure about rendering in views, but I'm guessing the sanitizer is going over this.

File: app/views/participant_bulk_actions/add_note.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<h2>Add Note</h2>
+<p>Add a note for <%= raw(@participants.collect{ |p| link_to p.fullname.html_safe, p }.to_sentence) %>.</p>
```
Comment (mitchellharper12) [inline app/views/participant_bulk_actions/add_note.html.erb]: I think this is possible XSS, as html_safe just changes the state of the string to be "html safe", but p.fullname hasn't gone through any sanitizing/escaping.

File: app/views/participant_bulk_actions/assign_mentor.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<h2>Assign to <%= Customer.mentor_Label %></h2>
+<p>Assign <%= raw(@participants.collect{ |p| link_to p.fullname.html_safe, p }.to_sentence) %> to the following <%= Customer.mentor_label %>:</p>
```
Comment (mitchellharper12) [inline app/views/participant_bulk_actions/assign_mentor.html.erb]: Same thing here as in `add_note.html.erb`

File: app/views/participant_bulk_actions/check_in_event.html.erb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<h2>Event Check-in</h2>
+<p>Check in <%= raw(@participants.collect{ |p| link_to p.fullname.html_safe, p }.to_sentence) %> to the following event:</p>
```
Comment (mitchellharper12) [inline app/views/participant_bulk_actions/check_in_event.html.erb]: Here as well

File: app/views/participant_bulk_actions/add_note.html.erb (hunk 3 above)
Comment (mattharris5) [inline app/views/participant_bulk_actions/add_note.html.erb]: Good catch!! Yikes. I am uploading fixes...

**Comment by mitchellharper12** [comment]
Everything looks great, this might be the most exciting PR in my time working on Dream Project, just for the Navbars for everyone.

File: app/views/participant_bulk_actions/check_in_event.html.erb (hunk 5 above)
Comment (mitchellharper12) [inline app/views/participant_bulk_actions/check_in_event.html.erb]: Just checking to see if you're updating this one separately

**Comment by mitchellharper12** [comment]
OK I JUST SAW THE BULK ADD MENTEES TO MENTOR FUNCTION AND ALSO ALL THE
OTHER BULK OPTIONS AND WITH THE NAVBAR FOR EVERYONE PEOPLE ARE GOING TO DIE
THIS IS THE GREATEST PR IN HISTORY THREE CHEERS THANKS A MILLION SERIOUSLY
THIS IS GOING TO CHANGE LIVES

**Pull request merged**